Objects/genobject.c

```diff
@@ -203,13 +203,15 @@ gen_send_ex(PyGenObject *gen, PyObject *arg, int exc, int closing)
     assert(f->f_back == NULL);
     f->f_back = tstate->frame;
 
-    if (exc) {
-        _PyErr_ChainStackItem(&gen->gi_exc_state);
-    }
-
     gen->gi_running = 1;
     gen->gi_exc_state.previous_item = tstate->exc_info;
     tstate->exc_info = &gen->gi_exc_state;
+
+    /* XXX: can we assert that _PyErr_Occurred() is true when exc is true? */
```

Comment:
I'd add an assertion and if the tests pass merge it. We can relax the check if any regression is found during the 3.9 release process--in which case we'd know the exact scenario and would be able to test it.